python/ray/tune/schedulers/pbt.py

```diff
@@ -231,8 +231,14 @@ def on_trial_add(self, trial_runner, trial):
         self._trial_state[trial] = PBTTrialState(trial)
 
     def on_trial_result(self, trial_runner, trial, result):
-        if self._time_attr not in result or self._metric not in result:
-            return TrialScheduler.CONTINUE
+        if self._time_attr not in result:
+            raise TuneError("Cannot find time_attr {} in trial results. "
```

Comment:
ValueError?

Reply:
wouldn't TuneError be preferable since it's more specific?

Reply:
Actually, I don't think TuneError is more specific; we raise TuneError for a variety of things (i.e., a trial failing, or a sync failing, etc), and this particular "TuneError" doesn't help the application catch that/filter out the specific failure mode.

What we should actually be doing is providing more detail about what type of error is happening.

Reply:
On the other hand, providing a more specific "ValueError" doesn't seem to buy you that much - it's just a user error and they probably shouldn't be catching it in the first place.